test/e2e/specs/realrp/realrp.go

```diff
@@ -0,0 +1,117 @@
+		By(fmt.Sprintf("OAS resource group is %s", resourcegroup))
+
+		accts, err := cli.Accounts.ListByResourceGroup(ctx, resourcegroup)
+		Expect(err).NotTo(HaveOccurred())
+
+		for _, acct := range *accts.Value {
+			By(fmt.Sprintf("trying to read account %s", *acct.Name))
+			_, err := cli.Accounts.ListKeys(ctx, resourcegroup, *acct.Name)
+			Expect(err).To(HaveOccurred())
+		}
+	})
+
+	It("should not be possible for customer to mutate an osa scale set", func() {
+		resourcegroup, err := cli.OSAResourceGroup(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"), os.Getenv("AZURE_REGION"))
+		Expect(err).NotTo(HaveOccurred())
+		By(fmt.Sprintf("OSA resource group is %s", resourcegroup))
+
+		scaleSets, err := cli.ListScaleSets(ctx, resourcegroup)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(scaleSets).Should(HaveLen(3))
+
+		for _, scaleSet := range scaleSets {
+			vms, err := cli.ListScaleSetVMs(ctx, resourcegroup, *scaleSet.Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("trying to update the scale set capacity")
+			_, err = cli.VirtualMachineScaleSets.Update(ctx, resourcegroup, *scaleSet.Name, compute.VirtualMachineScaleSetUpdate{
+				Sku: &compute.Sku{
+					Capacity: to.Int64Ptr(int64(len(vms) + 1)),
+				},
+			})
+			Expect(err).To(HaveOccurred())
+
+			By("trying to update the scale set type")
+			_, err = cli.VirtualMachineScaleSets.Update(ctx, resourcegroup, *scaleSet.Name, compute.VirtualMachineScaleSetUpdate{
+				Sku: &compute.Sku{
+					Name: to.StringPtr("Standard_DS1_v2"),
+				},
+			})
+			Expect(err).To(HaveOccurred())
+
+			By("trying to update the scale set SSH key")
+			_, err = cli.VirtualMachineScaleSets.Update(ctx, resourcegroup, *scaleSet.Name, compute.VirtualMachineScaleSetUpdate{
+				VirtualMachineScaleSetUpdateProperties: &compute.VirtualMachineScaleSetUpdateProperties{
+					VirtualMachineProfile: &compute.VirtualMachineScaleSetUpdateVMProfile{
+						OsProfile: &compute.VirtualMachineScaleSetUpdateOSProfile{
+							LinuxConfiguration: &compute.LinuxConfiguration{
+								SSH: &compute.SSHConfiguration{
+									PublicKeys: &[]compute.SSHPublicKey{
+										{
+											Path:    to.StringPtr("/home/cloud-user/.ssh/authorized_keys"),
+											KeyData: to.StringPtr(fakepubkey),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			})
+
+			Expect(err).To(HaveOccurred())
+
+			By("trying to create scale set script extension")
+			_, err = cli.VirtualMachineScaleSetExtensions.CreateOrUpdate(ctx, resourcegroup, *scaleSet.Name, "test", compute.VirtualMachineScaleSetExtension{
+				VirtualMachineScaleSetExtensionProperties: &compute.VirtualMachineScaleSetExtensionProperties{
+					Type:     to.StringPtr("CustomScript"),
+					Settings: `{"fileUris":["https://raw.githubusercontent.com/Azure-Samples/compute-automation-configurations/master/automate_nginx.sh"],"commandToExecute":"./automate_nginx.sh"}`,
+				},
+			})
+			Expect(err).To(HaveOccurred())
+
```

Comment:
`By("rebooting the scale set vms")`

Reply:
I don't understand this comment @charlesakalugwu ?

Reply:
please disregard it. I "crossed it out" :)